SignalR-Typescript-webpack: VS 2019 updates, VS Code updates, code updates. (#16835)
aspnet/AspNetCore.Docs: aspnetcore/tutorials/signalr-typescript-webpack.md

Configure Visual Studio to look for npm in the *PATH* environment variable. By default, Visual Studio uses the version of npm found in its installation directory. Follow these instructions in Visual Studio:

> 1. Launch Visual Studio 2019. At the start window, select **Continue without code**.

Reviewer: ```suggestion
1. Launch Visual Studio. At the start window, select **Continue without code**.
```

VS 2019 is listed as a prereq above. For that reason, let's drop the version number here.

Reviewer 2: One less thing to update in the next version change. We never specify the version in the topic outside the prereq .

1. Navigate to **Tools** > **Options** > **Projects and Solutions** > **Web Package Management** > **External Web Tools**.
1. Select the *$(PATH)* entry from the list. Click the up arrow to move the entry to the second position in the list, and select **OK**.

    ![Visual Studio Configuration](signalr-typescript-webpack/_static/signalr-configure-path-visual-studio.png)

Visual Studio configuration is completed. It's time to create the project.

1. Use the **File** > **New** > **Project** menu option and choose the **ASP.NET Core Web Application** template. Select **Next**.
1. Name the project *SignalRWebPack*, and select **Create**.

> 1. Select *.NET Core* from the target framework drop-down, and select *ASP.NET Core 3.0* from the framework selector drop-down. Select the **Empty** template, and select **Create**.

Reviewer: ```suggestion
1. Select *.NET Core* from the target framework drop-down, and select *ASP.NET Core 3.1* from the framework selector drop-down. Select the **Empty** template, and select **Create**.
```

> Add the **Microsoft.TypeScript.MSBuild** package to your project:

Reviewer: ```suggestion
Add the `Microsoft.TypeScript.MSBuild` package to your project:
```

As a rule of thumb, code-fence package names to avoid localization.

> 1. In Solution Explorer (right pane). right-click the project node and choose **Manage NuGet Packages**. In the **Browse** tab, search for **Microsoft.TypeScript.MSBuild**, and then click **Install** on the right to install the package.

Reviewer: Solution Explorer is a UI element, so let's bold it.
```suggestion
1. In **Solution Explorer** (right pane), right-click the project node and select **Manage NuGet Packages**. In the **Browse** tab, search for `Microsoft.TypeScript.MSBuild`, and then click **Install** on the right to install the package.
```

Reviewer 2: Per style guide, avoid (). In this case it's not needed. This isn't a super beginning tutorial.

> Visual Studio adds the NuGet package under the **Dependencies** node in Solution Explorer, enabling TypeScript compilation in your project.

Reviewer: ```suggestion
Visual Studio adds the NuGet package under the **Dependencies** node in **Solution Explorer**, enabling TypeScript compilation in your project.
```

# [Visual Studio Code](#tab/visual-studio-code)

Run the following command in the **Integrated Terminal**:

```dotnetcli
dotnet new web -o SignalRWebPack
code -r SignalRWebPack
```

* The `dotnet new` command creates an empty ASP.NET Core web app in a *SignalRWebPack* directory.
* The `code` command opens the *SignalRWebPack* folder in the current instance of Visual Studio Code.

> Run the following .NET Core CLI command in the **Integrated Terminal**:

Reviewer: ```suggestion
Run the following .NET Core CLI command in the **Integrated Terminal**:

```

1. Install the required npm packages. Execute the following command from the project root:

    ```console

>     npm install -D -E clean-webpack-plugin@3.0.0 css-loader@3.4.2 html-webpack-plugin@3.2.0 mini-css-extract-plugin@0.9.0 ts-loader@6.2.1 typescript@3.7.5 webpack@4.41.5 webpack-cli@3.3.10

Reviewer: ```suggestion
    npm i -D -E clean-webpack-plugin@3.0.0 css-loader@3.4.2 html-webpack-plugin@3.2.0 mini-css-extract-plugin@0.9.0 ts-loader@6.2.1 typescript@3.7.5 webpack@4.41.5 webpack-cli@3.3.10
```

Shorten `install` to `i`. It's consistent with the other install command below.

   [!code-csharp[Startup](signalr-typescript-webpack/sample/3.x/Startup.cs?name=snippet_AddSignalR)]

> 1. Create a new directory, called *Hubs*, in the project root *SignalRWebPack/*. Its purpose is to store the SignalR hub, which is created in the next step.

Reviewer: ```suggestion
1. Create a new directory, named *Hubs*, in the project root *SignalRWebPack/*. Its purpose is to store the SignalR hub class, which is created in the next step.
```

1. Execute the following command at the project root:

    ```console
    npm i @microsoft/signalr @types/node
    ```

>     The preceding command installs the [SignalR TypeScript client](https://www.npmjs.com/package/@microsoft/signalr), which allows the client to send messages to the server.

Reviewer: ```suggestion
    The preceding command installs:
    
     * The [SignalR TypeScript client](https://www.npmjs.com/package/@microsoft/signalr), which allows the client to send messages to the server.
     * The TypeScript type definitions for Node.js, which enables compile-time checking of Node.js types.
```

The type definitions package is installed here too. I recommend adding an explanation similar to what I've provided above.
